### PR TITLE
[WTF-1438] Document new source variables (9.24)

### PR DESCRIPTION
**Pull request by jeffersonRibeiro**
Add documentation for source variables in text templates, conditional settings, and association widgets in 9.24.

### CODE REVIEW

**Comment by ConnorLand** [comment]
@jeffersonRibeiro  Is this the main improvement documented in this PR?

"A [snippet](/refguide/snippet/) can also function as a data container when it has at least one snippet parameter. Unlike other data containers, a snippet never exposes a `$currentObject` variable because it can represent multiple independent objects."

If so, we need to add a bullet about that improvement to the [9.24 release notes](https://github.com/mendix/docs/pull/5759).

**Comment by jeffersonRibeiro** [comment]
> @jeffersonRibeiro Is this the main improvement documented in this PR?
> 
> "A [snippet](/refguide/snippet/) can also function as a data container when it has at least one snippet parameter. Unlike other data containers, a snippet never exposes a `$currentObject` variable because it can represent multiple independent objects."
> 
> If so, we need to add a bullet about that improvement to the [9.24 release notes](https://github.com/mendix/docs/pull/5759).

No, the main improvement was that `attribute-based expressions` for input elements can now select entities from outside the default entity context. For example a grand-parent data container or a snippet parameter.

The same improvement was done for the `reference selector` and `input reference set selector` when selecting an attribute path.

Since the refguide for the affected widgets already mention the `refguide/data-widgets` and the current doc description for selecting attributes doesn't specify the previous limitation, the only change needed for the docs was to add a snippet as a data container when it has at least 1 snippet parameter.

File: content/en/docs/refguide/modeling/pages/data-widgets/_index.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ The Data containers category contains the following widgets:
 
     {{< figure src="/attachments/refguide/modeling/pages/data-widgets/list-view-example.png"   width="450"  >}}
 
+A [snippet](/refguide/snippet/) can also function as a data container when it has at least one snippet parameter. Unlike other data containers, a snippet never exposes a `$currentObject` variable because it can represent multiple independent objects.
```
Comment (ConnorLand) [inline content/en/docs/refguide/modeling/pages/data-widgets/_index.md:30]: Ah, so this is the main upgrade documented in this PR. Ok thank you, I will point the RN bullet to it.

Comment (ConnorLand) [reply content/en/docs/refguide/modeling/pages/data-widgets/_index.md:30]: @jeffersonRibeiro thanks for your explanation in the comment above.

**Review by ConnorLand — APPROVED** [review]

**Comment by ConnorLand** [comment]
Basically customers have more options with how their attribute based expressions can refer to snippet parameters and data entities. The docs that discuss data entities and snippets were mostly already correct, except some of the small changes as made in this PR. 

This is a minor change in the product that we do want to let customers know about, but it does not require substantive new docs. Existing documentation on entities/snippets still holds true. So we will just craft a bullet for the 9.24 RNs, and let that bullet be the main way in docs customers learn about this small improvement.

**Comment by tamelbrun** [comment]
> This is a minor change in the product that we do want to let customers know about, but it does not require substantive new docs. Existing documentation on entities/snippets still holds true. So we will just craft a bullet for the 9.24 RNs, and let that bullet be the main way in docs customers learn about this small improvement.

@ConnorLand The "Selecting Attributes" section under "Improvements" already contains the RNs for the related stores, so this additional bullet point is not necessary.

**Comment by ConnorLand** [comment]
> > This is a minor change in the product that we do want to let customers know about, but it does not require substantive new docs. Existing documentation on entities/snippets still holds true. So we will just craft a bullet for the 9.24 RNs, and let that bullet be the main way in docs customers learn about this small improvement.
> 
> @ConnorLand The "Selecting Attributes" section under "Improvements" already contains the RNs for the related stores, so this additional bullet point is not necessary.

Ah I see, thank you.

**Pull request merged**